src/readinglistbar.cpp

```diff
@@ -31,7 +31,7 @@ void ReadingListBar::setupList()
     listWidget->clear();
     for(auto& bookmark:bookmarks) {
         auto reader = library->getReader(QString::fromStdString(bookmark.getBookId()));
-        if (reader == nullptr)
+        if (reader == nullptr || !library->getBookIds().contains(QString::fromStdString(bookmark.getBookId())))
```

Comment:
When it is possible that the library find a reader for the book but in the same time the library doesn't contain the book ?

Reply:
After removing a zim

Reply:
But the `getReader` should return `nullptr` in this case no ?

Reply:
I think it should yes :)
I am still able to access to deleted zim with the readinglist (without this pr) is it related to this reader that is not null ?

Reply:
This is probably because the reader is still in the `m_readersMap` of the library.
We should remove it from the reader when we remove the book.

Reply:
And the new version of kiwix-lib also has an internal map of readers (I've add it for the integration of the server in the kiwix-lib). We should use it in kiwix-desktop instead of having a separated map.
But there is the same bug in kiwix-lib, the reader is not removed from the map when we remove the book.